### PR TITLE
Increase PHPStan to level 2

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is for the next major.

## Changelog

```markdown
### Added
- Add `PageManager::createWithDefaults()` to create Pages with some default values.
```

### CODE REVIEW

File: src/Admin/BlockAdmin.php [hunk 1]
```diff
@@ -231,20 +229,6 @@ private function configureBlockFields(FormMapper $form, BlockInterface $block):
             } else {
                 $service->configureCreateForm($blockMapper, $block);
             }
-        } else {
-            @trigger_error(
```
Comment (VincentLanglet) [inline src/Admin/BlockAdmin.php:233]: According to the deprecation you have to change the check ligne 207 to "instanceof Editable"

File: src/Admin/SharedBlockAdmin.php [hunk 2]
```diff
@@ -113,20 +113,6 @@ private function configureBlockFields(FormMapper $form, BlockInterface $block):
             } else {
                 $service->configureCreateForm($blockMapper, $block);
             }
-        } else {
-            @trigger_error(
```
Comment (VincentLanglet) [inline src/Admin/SharedBlockAdmin.php:115]: Same

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/Admin/BlockAdmin.php (hunk 1 above)
Comment (jordisala1991) [reply src/Admin/BlockAdmin.php:233]: Don't know what you mean, the if already have an instanceof check, right?

Comment (VincentLanglet) [reply src/Admin/BlockAdmin.php:233]: (Im on phone)

The code is

If not instanceof Blockeditableservice, throw exception Foo

If instanceof EditableBlock, things,
Else deprecation

This must become

If not instanceof EditableBlock,
Throw exception Foo

And then things

Dunno if Im clearer or ill need to access a computer next week

Comment (jordisala1991) [reply src/Admin/BlockAdmin.php:233]: I will try, I thing I understood you :)

Comment (jordisala1991) [reply src/Admin/BlockAdmin.php:233]: Fixed.

File: src/Admin/SharedBlockAdmin.php (hunk 2 above)
Comment (jordisala1991) [reply src/Admin/SharedBlockAdmin.php:115]: Fixed.

**Review by VincentLanglet — DISMISSED** [review]

**Comment by VincentLanglet** [comment]
Good to me (except tests)

File: src/Command/CreateSiteCommand.php [hunk 3]
```diff
@@ -68,6 +69,7 @@ public function configure(): void
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $helper = $this->getHelper('question');
```
Comment (VincentLanglet) [inline src/Command/CreateSiteCommand.php]: This should work with phpstan symfony extension if the console is configured

File: src/Admin/BaseBlockAdmin.php [hunk 4]
```diff
@@ -107,14 +96,8 @@ public function prePersist($object): void
         if ($object->getPage() instanceof PageInterface) {
             $object->getPage()->setEdited(true);
         }
-
-        // fix weird bug with setter object not being call
```
Comment (VincentLanglet) [inline src/Admin/BaseBlockAdmin.php:111]: We can remove this code ? There is a comment talking about a bug

Comment (jordisala1991) [reply src/Admin/BaseBlockAdmin.php:111]: there is also #1291

File: src/Command/CreateSiteCommand.php (hunk 3 above)
Comment (jordisala1991) [reply src/Command/CreateSiteCommand.php]: I will try to configure it.

Comment (jordisala1991) [reply src/Command/CreateSiteCommand.php]: Fixed,

**Review by VincentLanglet — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**